### PR TITLE
Using Mapbox map on csedweek.org homepage

### PR DESCRIPTION
**Pull request by daynew**
# Description
csedweek.org displays a map of all the hour-of-code events happening around
the world. This map was using Google Fusion Table which will soon be
turned off by Google. With this change, csedweek.org will now use a
Mapbox map to display the hour-of-code events.

## Links
- [Hour Of Code Homepage](https://hourofcode.com)
- [Current CsEdWeek Homepage](https://csedweek.org)
- [jira](https://codedotorg.atlassian.net/browse/FND-748)

## Testing story
* Manually verified on (Linux) X (Firefox, Chrome)
* `cd pegasus; bundle exec rake test`

### Before
Kazam_screenshot_00005

### After
Kazam_screenshot_00006

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: pegasus/sites.v3/csedweek.org/public/css/map.css [hunk 1]
```diff
@@ -41,15 +59,14 @@
   float: left;
 }
 
-.blue {
-  background: #5781fc;
+.red {
```
Comment (Hamms) [inline pegasus/sites.v3/csedweek.org/public/css/map.css]: are we required to use these classes, or can we use something a little more descriptive? http://thecodelesscode.com/case/95

File: pegasus/sites.v3/csedweek.org/public/index.haml [hunk 2]
```diff
@@ -48,7 +47,20 @@ video_player: true
...
+      %a{:href=> "https://hourofcode.com/events"}
+        = I18n.t(:csedweek_map_header).gsub('#', format_integer_with_commas(fetch_hoc_metrics['total_hoc_count']))
+      %a{:href=> "https://hourofcode.com/#join"}
+        %button.signup-button
+          = I18n.t(:host_an_hour)
+    = view :map
+    .row
+      .col-sm-12
+        .footnote
+          The Hour of Code map only shows the first 200,000 registered events. All registered events with a valid address can be viewed on
```
Comment (Hamms) [inline pegasus/sites.v3/csedweek.org/public/index.haml]: is this still true? I thought part of the advantage of mapbox was that we were no longer restricted like this.

File: pegasus/sites.v3/csedweek.org/public/index.haml [hunk 3]
```diff
@@ -48,7 +47,20 @@ video_player: true
 
   %br
   %div{:style=>'clear:both;'}
-  = view :map
+  #hocmap
+    %h2.mapheader
+      %a{:href=> "https://hourofcode.com/events"}
+        = I18n.t(:csedweek_map_header).gsub('#', format_integer_with_commas(fetch_hoc_metrics['total_hoc_count']))
+      %a{:href=> "https://hourofcode.com/#join"}
```
Comment (Hamms) [inline pegasus/sites.v3/csedweek.org/public/index.haml]: should these be using `resolve_url`?

**Review by Hamms — COMMENTED** [review]
Why was so much map-specific markup moved from the `map` view into `index`?

**Comment by dju90** [comment]
Can we put this map behind an experiment flag as well, just like on code.org/yourschool? I would like to have at least a day or two to have folks in the org load this page to make sure there are no issues.

File: pegasus/sites.v3/csedweek.org/public/index.haml (hunk 2 above)
Comment (daynew) [reply pegasus/sites.v3/csedweek.org/public/index.haml]: I have no idea. I'm waiting for a response from Bethany.

**Comment by daynew** [comment]
> Can we put this map behind an experiment flag as well, just like on code.org/yourschool? I would like to have at least a day or two to have folks in the org load this page to make sure there are no issues.

As we talked about offline, since the map is already well tested on hourofcode.com, we will just make sure to test on staging before deploying to production.

File: pegasus/sites.v3/csedweek.org/public/index.haml (hunk 2 above)
Comment (daynew) [reply pegasus/sites.v3/csedweek.org/public/index.haml]: Bethany confirmed that Mapbox does allow us to show all the registered events now. Good catch.

**Comment by daynew** [comment]
> Why was so much map-specific markup moved from the `map` view into `index`?

This was just some early refactoring I did when playing around with the code. I moved it back into map.haml

File: pegasus/sites.v3/csedweek.org/public/css/map.css [hunk 4]
```diff
@@ -28,11 +33,24 @@
 }
 
 .legend-title {
-  font-size: 18px;
-  color: rgb(193, 43, 39);
+	font-size: 18px;
```
Comment (Hamms) [inline pegasus/sites.v3/csedweek.org/public/css/map.css]: nit: overindented

File: pegasus/sites.v3/csedweek.org/views/map.haml [hunk 5]
```diff
@@ -1,182 +1,32 @@
...
+    .color.legend-special-event
+    %div= hoc_s(:map_legend_cs_tech_jam)
+
   .row
     .col-sm-12
-      #gmap.full-width
       .footnote
-        The Hour of Code map only shows the first 200,000 registered events. All registered events with a valid address can be viewed on
-        %a{href: resolve_url('/events')}
+        All registered events with a valid address can be viewed on
```
Comment (Hamms) [inline pegasus/sites.v3/csedweek.org/views/map.haml:26]: can we add a follow-up work item to translate this string?

**Review by Hamms — APPROVED** [review]
this looks great!

File: pegasus/sites.v3/csedweek.org/views/map.haml (hunk 5 above)
Comment (dju90) [reply pegasus/sites.v3/csedweek.org/views/map.haml:26]: https://codedotorg.atlassian.net/browse/FND-868

**Pull request merged**